cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -828,74 +838,74 @@ createDelegateKeys :: FilePath -> Word -> ExceptT ShelleyGenesisCmdError IO ()
 createDelegateKeys dir index = do
   liftIO $ createDirectoryIfMissing False dir
   runGenesisKeyGenDelegate
-        (VerificationKeyFile $ dir </> "delegate" ++ strIndex ++ ".vkey")
-        coldSK
-        opCertCtr
+        (File @OfVerificationKey $ dir </> "delegate" ++ strIndex ++ ".vkey")
+        (onlyOut coldSK)
+        (onlyOut opCertCtr)
   runGenesisKeyGenDelegateVRF
-        (VerificationKeyFile $ dir </> "delegate" ++ strIndex ++ ".vrf.vkey")
-        (SigningKeyFile $ dir </> "delegate" ++ strIndex ++ ".vrf.skey")
+        (File @OfVerificationKey $ dir </> "delegate" ++ strIndex ++ ".vrf.vkey")
+        (File @OfSigningKey $ dir </> "delegate" ++ strIndex ++ ".vrf.skey")
   firstExceptT ShelleyGenesisCmdNodeCmdError $ do
     runNodeKeyGenKES
-        kesVK
-        (SigningKeyFile $ dir </> "delegate" ++ strIndex ++ ".kes.skey")
+        (onlyOut kesVK)
+        (File @OfSigningKey $ dir </> "delegate" ++ strIndex ++ ".kes.skey")
     runNodeIssueOpCert
-        (VerificationKeyFilePath kesVK)
-        coldSK
+        (VerificationKeyFilePath (onlyIn kesVK))
+        (onlyIn coldSK)
         opCertCtr
         (KESPeriod 0)
-        (OutputFile $ dir </> "opcert" ++ strIndex ++ ".cert")
+        (File $ dir </> "opcert" ++ strIndex ++ ".cert")
  where
    strIndex = show index
-   kesVK = VerificationKeyFile $ dir </> "delegate" ++ strIndex ++ ".kes.vkey"
-   coldSK = SigningKeyFile $ dir </> "delegate" ++ strIndex ++ ".skey"
-   opCertCtr = OpCertCounterFile $ dir </> "delegate" ++ strIndex ++ ".counter"
+   kesVK = File @OfVerificationKey $ dir </> "delegate" ++ strIndex ++ ".kes.vkey"
+   coldSK = File @OfSigningKey $ dir </> "delegate" ++ strIndex ++ ".skey"
```

Comment:
`coldSK` is a signing key file that is both written to and read from in the same command.  It is therefore annotated as `InOut`. This type is inferred, but can be made explicit with type application of `@InOut`.